src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -80,70 +89,184 @@ def __init__(self, env_layer, execution_config, composite_logger, telemetry_writ
+                    if file_name.endswith(self.APT_SOURCES_LIST_DIR_LIST_EXT):    # .list type
+                        source_parts_list_content += "\n" + self.__read_one_line_style_list_format(file_path, max_patch_published_date, base_classification)
+                    elif file_name.endswith(self.APT_SOURCES_LIST_DIR_SRC_EXT):   # .sources type
+                        source_parts_deb882_style_content += "\n" + self.__read_deb882_style_format(file_path, max_patch_published_date, base_classification)
+
+                except Exception as error:      # does not throw to allow patching to happen with functioning sources
+                    self.composite_logger.log_error("[APM] Error while processing consolidated sources list. [File={0}][Error={1}]".format(file_name, repr(error)))
+
+        return source_parts_deb882_style_content.strip(), source_parts_list_content.strip()
+
+    def __read_one_line_style_list_format(self, file_path, max_patch_published_date, base_classification):
+        # type: (str, str, str) -> str
+        # Reads a *.list file and returns only lines that are functionally required
+        self.composite_logger.log_verbose("[APM] Reading source list for consolidation. [FilePath={0}]".format(file_path))
+
+        std_source_list_content = str()
+        sources_content = self.env_layer.file_system.read_with_retry(file_path)
+        self.composite_logger.log_verbose("[APM][Srclist] Input source list file. [FilePath={0}][Content={1}]".format(file_path, sources_content))
+        sources_content_lines = sources_content.splitlines()
+
+        for line in sources_content_lines:
+            if len(line.strip()) != 0 and not line.strip().startswith("#"):
+                if base_classification == Constants.PackageClassification.SECURITY and "security" not in line:
+                    continue
+                std_source_list_content += "\n" + self.__apply_max_patch_publish_date(sources_content=line, max_patch_publish_date=max_patch_published_date)
+
+        self.composite_logger.log_verbose("[APM][Srclist] Output source list slice. [FilePath={0}][TransformedContent={1}]".format(file_path, std_source_list_content))
+        return std_source_list_content.strip()
+
+    def __read_deb882_style_format(self, file_path, max_patch_published_date, base_classification):
+        # type: (str, str, str) -> str
+        std_source_parts_content = str()
+        stanza = str()
+
+        try:
+            source_parts_content = self.env_layer.file_system.read_with_retry(file_path) + "\n"
+            self.composite_logger.log_verbose("[APM][Deb882] Input source parts file. [FilePath={0}][Content={1}]".format(file_path, source_parts_content))
+            source_parts_content_lines = source_parts_content.splitlines()
+
+            for line in source_parts_content_lines:
+                if line.startswith("#"):    # comments
+                    continue
+
+                if len(line.strip()) == 0:  # stanza separating line
```

Comment:
we should have sample code here, to help understand how and what a stanza includes

Reply:
Will address with extra file(s)

Reply:
Addressed